driver-core/src/main/java/com/datastax/driver/core/ArrayBackedRow.java

```diff
@@ -239,6 +242,23 @@ public InetAddress getInet(String name) {
         return getInet(metadata.getFirstIdx(name));
     }
 
+    public Token getToken(int i) {
+        if (tokenFactory == null)
+            throw new DriverInternalError("Token factory not set. This should only happen at initialization time");
+
+        metadata.checkType(i, tokenFactory.getTokenType().getName());
+
+        ByteBuffer value = data.get(i);
+        if (value == null || value.remaining() == 0)
+            return null;
+
+        return tokenFactory.deserialize(value);
+    }
+
+    public Token getToken(String name) {
+        return getToken(metadata.getFirstIdx(name));
```

Comment:
As an alternative to the magic constant, we could have dedicated methods (`getPartitionKeyToken()` and `setPartitionKeyToken(Token)`).

For the getter (whether it's magic constant or not), we'll have to iterate over the variable names and pick the first one that matches `token(...)`, we can't do much better because we don't have access to table schema at this point.

I think it's worth leaving the "i => token(i)" logic in the named getter.